### PR TITLE
Update launcher table in the documentation

### PR DESCRIPTION
**Pull request by jhh67**
Update the launcher table to include missing launchers and standardize the language.

### CODE REVIEW

File: doc/rst/usingchapel/launcher.rst [hunk 1]
```diff
@@ -341,3 +342,6 @@ Launcher Name  Description
 =============  ==========================================================
 mpirun         launch using mpirun (no mpi comm currently) 
 =============  ==========================================================
+
+.. |-| unicode:: U+2011 .. non-breaking hyphen
+  :trim:
```
Comment (ronawho) [inline doc/rst/usingchapel/launcher.rst:347]: This seems a little unfortunate in that it makes the raw text harder to read or copy+paste, but I agree it makes the rendered table a little nicer.

Comment (jhh67) [reply doc/rst/usingchapel/launcher.rst:347]: I didn't think about the copy+paste issue. Do you think it's worth doing?

Comment (ronawho) [reply doc/rst/usingchapel/launcher.rst:347]: Probably the rendered text is the most important thing, so I learn towards keeping it.

**Comment by ronawho** [comment]
I'm ok going with this change, but wanted to note that I think a potential benefit of the old styles was that it focused more on the user-facing aspects of the hardware/system. Like "oh, I'm on an InfiniBand system with SLURM workload manager, I should use `slurm-gasnetrun_ibv`. And it is "Am I using the ibv substrate.. oh, yeah that's what I have set, so I should use that"

I think this is more precise so am fine with the change, just noting if shifts the user perspective from the system to the chplenv configuration.

**Review by ronawho — APPROVED** [review]

**Pull request merged**